packages/workspace/src/browser/workspace-service.ts

```diff
@@ -21,21 +21,28 @@ import { FileSystemWatcher } from '@theia/filesystem/lib/browser';
 import { WorkspaceServer } from '../common';
 import { WindowService } from '@theia/core/lib/browser/window/window-service';
 import { FrontendApplication, FrontendApplicationContribution } from '@theia/core/lib/browser';
+import { Disposable, Emitter, Event } from '@theia/core/lib/common';
 import { Deferred } from '@theia/core/lib/common/promise-util';
 import { ILogger } from '@theia/core/lib/common/logger';
 import { WorkspacePreferences } from './workspace-preferences';
 
+export interface WorkspaceDescription {
+    workspaceFolder: FileStat | undefined;
+    roots: FileStat[]
+}
+
 /**
  * The workspace service.
  */
 @injectable()
 export class WorkspaceService implements FrontendApplicationContribution {
 
-    private _root: FileStat | undefined;
+    private _workspaceFolder: FileStat | undefined;
```

Comment:
Isn't this pointing either to a folder or a description file? So it should not be called 'folder' if it can point to a file.

Reply:
we should remove it and use `_workspaceDescription.workspaceFolder` instead.